src/toolbox.js

```diff
@@ -113,4 +120,26 @@ var Toolbox = GObject.registerClass({
     shutdown() {
         void this;
     }
+
+    _updateResetButton() {
+        let visible = false;
+        try {
+            const [variant] = this._gameState.GetSync(RESET_BUTTON_KEY);
+            visible = variant.deep_unpack()
+                             .visible
+                             .deep_unpack();
+        } catch (e) {
+            logError(e);
+        }
+        this._buttonReset.visible = visible;
+    }
+
+    _onStateChanged(proxy, sender, signal, params) {
+        if (signal !== 'changed')
+            return;
+        const key = params.get_child_value(0).deep_unpack();
+        if (key !== RESET_BUTTON_KEY)
+            return;
+        this._updateResetButton();
```

Comment:
Or alternatively, get the value from `get_child_value(1)` and update the state based on that. It just seems wasteful to do another sync call if you already have the value here.

Reply:
Yeah, this is what I meant. But, I am curious about how you approach works.